test/retransmission_test.exs

```diff
@@ -4,68 +4,57 @@ defmodule ExDTLS.RetransmissionTest do
   test "check retransmission" do
     {:ok, rx_pid} = ExDTLS.start_link(client_mode: false, dtls_srtp: true)
     {:ok, tx_pid} = ExDTLS.start_link(client_mode: true, dtls_srtp: true)
-    {:ok, packets} = ExDTLS.do_handshake(tx_pid)
-    assert :ok == loop({rx_pid, false}, {tx_pid, false}, packets)
-  end
 
-  @tag timeout: :infinity
-  @tag :capture_log
-  @tag :long_running
-  test "check reaching timeout limit" do
-    {:ok, rx_pid} = ExDTLS.start_link(client_mode: false, dtls_srtp: true)
-    {:ok, tx_pid} = ExDTLS.start_link(client_mode: true, dtls_srtp: true)
-    {:ok, packets} = ExDTLS.do_handshake(tx_pid)
+    ExDTLS.do_handshake(tx_pid)
+    packets = wait_for_retransmission(tx_pid)
 
-    Process.flag(:trap_exit, true)
+    ExDTLS.process(rx_pid, packets)
+    packets = wait_for_retransmission(rx_pid)
 
-    try do
-      process_until_raise(rx_pid, packets, 1)
-    catch
-      :exit, _msg ->
-        assert true
-    end
+    ExDTLS.process(tx_pid, packets)
+    packets = wait_for_retransmission(tx_pid)
+
+    assert finish_hsk(rx_pid, tx_pid, packets) == :ok
   end
 
-  defp loop({pid1, false}, {pid2, state2}, packets) do
-    case process_second_message(pid1, packets) do
-      {:retransmit, _component, packets} -> loop({pid2, state2}, {pid1, true}, packets)
-      :error -> :error
+  defp wait_for_retransmission(pid, first? \\ true) do
+    receive do
+      1000 ->
+        wait_for_retransmission(pid, false)
+
+      {:retransmit, ^pid, packets} ->
+        packets
+
+      _other ->
+        wait_for_retransmission(pid, first?)
     end
   end
 
-  defp loop({pid1, true}, {pid2, true}, packets) do
+  defp finish_hsk(pid1, pid2, packets) do
     {:handshake_finished, _handshake_data, packets} = ExDTLS.process(pid1, packets)
+
     {:handshake_finished, _handshake_data} = ExDTLS.process(pid2, packets)
+
     :ok
   end
 
-  defp process_second_message(pid, packets) do
-    case ExDTLS.process(pid, packets) do
-      _msg -> nil
-    end
-
-    receive do
-      1_000 ->
-        :error
-
-      msg ->
-        msg
-    end
-  end
+  @tag :long_running
+  @tag timeout: :infinity
+  test "check reaching timeout limit" do
+    Process.flag(:trap_exit, true)
```

Comment:
Is it still needed?

Reply:
It is needed because without it test will fail of TimeoutError